### PR TITLE
Migrate Windows ci to Github action for python-only

### LINKED ISSUE #2654: Migrate Windows CI: Migrate Remaining Python test

### PR DESCRIPTION
**Pull request by FirelightFlagboy**
closes #2654

### CODE REVIEW

File: .github/workflows/windows-ci.yml [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+          cache: poetry
+
+      - name: Install python deps
+        run: poetry install -E core -E backend
+
+      - name: Generate PyQT GUI data
+        run: poetry run python misc/generate_pyqt.py
+
+      - name: Test fast
+        run: poetry run pytest tests
```
Comment (touilleMan) [inline .github/workflows/windows-ci.yml:51]: Should use `-n auto` option when running the tests to enable parallelism

File: .github/workflows/windows-ci.yml [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+env:
+  rust-version: 1.62.0
+  python-version: "3.9"
+  poetry-version: 1.1.13
+  sqlite-version: 3390100
+  winfsp-version: 1.8.20304
+
+jobs:
+  python-only:
+    runs-on: windows-2019
```
Comment (touilleMan) [inline .github/workflows/windows-ci.yml:18]: Why 2019 ? Why not the latest windows?

Comment (FirelightFlagboy) [reply .github/workflows/windows-ci.yml:18]: I've got some problems with `windows-2022` on the rust part when installing `diesel-rs`, but I could try to use it for the python part

File: .github/workflows/windows-ci.yml (hunk 1 above)
Comment (FirelightFlagboy) [reply .github/workflows/windows-ci.yml]: done

**Review by touilleMan — APPROVED** [review]

File: .github/workflows/windows-ci.yml [hunk 3]
```diff
@@ -0,0 +1,64 @@
...
+          cache: poetry
+
+      - name: Install python deps
+        run: poetry install -E core -E backend
+
+      - name: Generate PyQT GUI data
+        run: poetry run python misc/generate_pyqt.py
+
+      - name: Test fast
+        run: poetry run pytest tests ${{ env.pytest-args }} -n auto
```
Comment (touilleMan) [inline .github/workflows/windows-ci.yml:55]: Can you use `-x` also on fast tests ? this makes the build fail faster and avoids huge error log when something is definitely wrong (e.g. https://github.com/Scille/parsec-cloud/runs/7437053385?check_suite_focus=true)

Comment (FirelightFlagboy) [reply .github/workflows/windows-ci.yml:56]: done in 4c64024

**Review by touilleMan — APPROVED** [review]

**Pull request merged**